data/h3_raster_importer/tiff_folder_to_h3_table.py

```diff
@@ -0,0 +1,145 @@
+"""
+
+import os
+from io import StringIO
+import re
+from math import ceil
+
+import psycopg2
+import h3ronpy.raster
+import numpy as np
+import pandas as pd
+import rasterio as rio
+import logging
+from docopt import docopt
+
+
+DTYPES_TO_PG = {
+    'boolean': 'bool',
+    'uint8': 'smallint',
+    'uint16': 'int',
+    'uint32': 'bigint',
+    'uint64': 'bigint',
+    'int8': 'smallint',
+    'int16': 'smallint',
+    'int32': 'bigint',
+    'int64': 'bigint',
+    'float32': 'real',
+    'float64': 'double precision'
+}
+BLOCKSIZE=512
+
+logging.basicConfig(level=logging.INFO)
+
+def slugify(s):
+    return re.sub("[^0-9a-z_]+", "_", s.lower())
+
+
+def gen_raster_h3(raster_list, h3_res, geo=False):
+    """Convert a list of identically formatted rasters to H3
+    
+    A function for efficiently turning a set of rasters into an H3 table.
+    
+    Takes a list of 1-band rasters with identical projection/transform.
+    Reads each raster in blocks, and converts to h3 (nearest to centroid).
+    Yields a dataframe with an h3index and one column for each raster's value.
+
+    Args:
+        raster_list: list of paths to rasters
+        h3_res: h3 resolution to use for resampling
+    
+    Yields:
+        A Pandas dataframe for each raster block (usu. 512x512) with an 
+        h3index and one column for each raster's value.
+    """
+    readers = [rio.open(r) for r in raster_list]
+    names = [slugify(os.path.splitext(os.path.basename(r))[0]) for r in raster_list]
+    
+    base = readers[0]
+    for j in range(ceil(base.height/BLOCKSIZE)):
+        for i in range(ceil(base.width/BLOCKSIZE)):
+            window = rio.windows.Window(i*BLOCKSIZE, j*BLOCKSIZE, BLOCKSIZE, BLOCKSIZE)
+            w_transform = rio.windows.transform(window, base.transform)
+            dfs = []
+            for src in readers:
+                if src.transform != base.transform:
+                    raise ValueError("Transforms do not match")
+                arr = src.read(1, window=window)
+                _df = h3ronpy.raster.raster_to_dataframe(arr, w_transform, h3_res, nodata_value=src.profile['nodata'], compacted=False, geo=geo)
+                dfs.append(_df.set_index('h3index')['value'])
+            df = pd.concat(dfs, axis=1)
+            logging.info(f'Reading block {j}, {i}: h3index count {len(df)}')
+            if len(df):
+                df.columns = names
+                df.index = pd.Series(df.index).apply(lambda x: hex(x)[2:])
+                # cast h3index from int64 to hex string
+                yield df
+    for src in readers:
+        src.close()
+
+
+def load_raster_list_to_h3_table(raster_list, table, h3_res):
+    conn = psycopg2.connect(
+        host=os.getenv('API_POSTGRES_HOST'),
+        port=os.getenv('API_POSTGRES_PORT'),
+        user=os.getenv('API_POSTGRES_USERNAME'),
+        password=os.getenv('API_POSTGRES_PASSWORD')
+    )
+    cursor = conn.cursor()
+    cursor.execute(f"DROP TABLE IF EXISTS {table};")
```

Comment:
I believe the API should take care of the existence of the table, as it already takes care of generating the full data-model on the DB at init, also this is relevant for us in order to access the data stored on said table